sendgrid.go

```diff
@@ -26,19 +26,38 @@ type SGClient struct {
 }
 
 // NewSendGridClient will return a new SGClient.
-func NewSendGridClient(apiUser, apiPwd string) *SGClient {
+func NewSendGridClient(apiUserOrKey string, apiKey ...string) *SGClient {
```

Comment:
I realize I could have just done something like

```go
func NewSendGridClient(authParams ...string) *SGClient {
```

Is that still too magical? Go the route of the 2nd constructor?